lambdas/call-ticf-cri/src/main/java/uk/gov/di/ipv/core/callticfcri/CallTicfCriHandler.java

```diff
@@ -2,17 +2,173 @@
+        this.ticfCriService = new TicfCriService(configService);
+        this.ciMitService = new CiMitService(configService);
+        this.criStoringService =
+                new CriStoringService(
+                        configService,
+                        new AuditService(AuditService.getDefaultSqsClient(), configService),
+                        null,
+                        new VerifiableCredentialService(configService),
+                        ciMitService);
+    }
+
+    public CallTicfCriHandler(
+            ConfigService configService,
+            IpvSessionService ipvSessionService,
+            ClientOAuthSessionDetailsService clientOAuthSessionDetailsService,
+            UserIdentityService userIdentityService,
+            TicfCriService ticfCriService,
+            CiMitService ciMitService,
+            CriStoringService criStoringService) {
+        this.configService = configService;
+        this.ipvSessionService = ipvSessionService;
+        this.clientOAuthSessionDetailsService = clientOAuthSessionDetailsService;
+        this.userIdentityService = userIdentityService;
+        this.ticfCriService = ticfCriService;
+        this.ciMitService = ciMitService;
+        this.criStoringService = criStoringService;
+    }
+
     @Override
     @Tracing
     @Logging(clearState = true)
-    public Map<String, Object> handleRequest(JourneyRequest input, Context context) {
-        return null;
+    public Map<String, Object> handleRequest(ProcessRequest input, Context context) {
+        LogHelper.attachComponentIdToLogs(configService);
+        LogHelper.attachCriIdToLogs(TICF_CRI);
+
+        try {
+            String ipvSessionId = RequestHelper.getIpvSessionId(input);
+            String featureSet = RequestHelper.getFeatureSet(input);
+            String journeyType = RequestHelper.getJourneyType(input);
+            configService.setFeatureSet(featureSet);
+            IpvSessionItem ipvSessionItem = ipvSessionService.getIpvSession(ipvSessionId);
+            ClientOAuthSessionItem clientOAuthSessionItem =
+                    clientOAuthSessionDetailsService.getClientOAuthSession(
+                            ipvSessionItem.getClientOAuthSessionId());
+            LogHelper.attachGovukSigninJourneyIdToLogs(
+                    clientOAuthSessionItem.getGovukSigninJourneyId());
+
+            List<SignedJWT> ticfVcs =
+                    ticfCriService.getTicfVc(
+                            clientOAuthSessionItem,
+                            ipvSessionItem,
+                            journeyType.equals(REUSE_JOURNEY_TYPE)
+                                    ? List.of()
+                                    : userIdentityService.getUserIssuedCredentials(
+                                            clientOAuthSessionItem.getUserId()));
+
+            if (ticfVcs.isEmpty()) {
+                LOGGER.info("No VC to process - returning next");
+                return JOURNEY_NEXT;
```

Comment:
I think this is probably an error - SPOT is likely to fail us if we don't send a TICF VC